internal/util/flowgraph/node.go

```diff
@@ -53,26 +53,156 @@ type BaseNode struct {
 	maxParallelism int32
 }
 
+// manage nodeCtx
+type nodeCtxManager struct {
+	inputNodeCtx *nodeCtx
+	closeWg      *sync.WaitGroup
```

Comment:
passing `WaitGroup` into a struct is not a good idea, add 2 `StartXxx` methods and operates the `WaitGroup` in outside

Reply:
WaitGroup is fg.closeWg, fg.close will do this.